hooks/useENS.ts

```diff
@@ -0,0 +1,31 @@
+import { ethers } from 'ethers';
+import { useEffect, useState } from 'react';
+
+const useENS = (address?: string): { ensName: string | null; ensAvatar: string | null } => {
```

Comment:
I'd love to see this take a list of addresses instead of a single address. This way, we could call the function any time our list of addresses in the leaderboard changes, and return an ENS object with the name and avatar for each address. It would reduce the number of calls to the hook by hundreds with the same result.

Reply:
I'm open to that if we can be sure that we are only making the ENS call, and not doing additional stuff like creating a new provider on every call.

Reply:
I will try to use a signer you mentioned.

Reply:
I just found that there is a `staticMainnetProvider` that can be used.